### PR TITLE
Replace `return; yield` with `yield from []`

### PR DESCRIPTION
**Pull request by ruudk**
This has the same effect, but looks less hacky and makes PHPStan happy.
https://phpstan.org/r/df7fcc88-1df8-428e-b675-5dc6965c34d6

Previously this was needed to work properly with output capturing.

/cc @fabpot @nicolas-grekas 

https://3v4l.org/KXste

### CODE REVIEW

**Comment by derrabus** [comment]
I think, the goal is to have a generator function which is not executed until we iterate over the result. This does not happen if you remove all `yield` instructions and return an empty array. But we could do `yield from [];` instead.

**Comment by ruudk** [comment]
> This does not happen if you remove a `yield` instructions and return an empty array

Are you sure? https://3v4l.org/GrX71

**Comment by ruudk** [comment]
> we could do `yield from [];` instead

I didn't think of this. I like it as well, as this will make sure the return type is always a Generator.

**Comment by ruudk** [comment]
Currently, PHPStan does not complain about this 👏 
https://phpstan.org/r/df7fcc88-1df8-428e-b675-5dc6965c34d6

**Comment by ruudk** [comment]
I rebased the PR and targeted v3 and applied the `yield from []` trick suggested by @derrabus.

File: src/Node/CaptureNode.php [hunk 1]
```diff
@@ -36,10 +36,10 @@ public function compile(Compiler $compiler): void
         }
         $compiler
             ->raw($useYield ? "implode('', iterator_to_array(" : '\\Twig\\Extension\\CoreExtension::captureOutput(')
-            ->raw("(function () use (&\$context, \$macros, \$blocks) {\n")
+            ->raw("(function () use (&\$context, \$macros, \$blocks): iterable {\n")
```
Comment (nicolas-grekas) [inline src/Node/CaptureNode.php]: this is not needed, generated code don't need extra runtime checks, let's remove them all from this PR

**Review by nicolas-grekas — COMMENTED** [review]
looks nicer indeed

**Comment by derrabus** [comment]
> > This does not happen if you remove a `yield` instructions and return an empty array
> 
> Are you sure? https://3v4l.org/GrX71

This reproducer misses the point because the function call and the start of the iteration are on the same line.

see https://3v4l.org/78LhS

**Review by nicolas-grekas — APPROVED** [review]

File: src/Node/CaptureNode.php (hunk 1 above)
Comment (ruudk) [reply src/Node/CaptureNode.php]: Done!

**Comment by ruudk** [comment]
@nicolas-grekas You approved and I requested review at roughly the same time 😅

**Review by derrabus — APPROVED** [review]

**Review by xabbuh — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ruudk.

**Pull request merged**